src/bin/krimedc.rs

```diff
@@ -330,13 +330,15 @@ enum Opt {
 struct UserPrincipal {
     name: String,
     password: String,
+    kvno: u32,
```

Comment:
So this in a nice trick: Because this is serde with Deserialise, you can give this field an attribute from [here](https://serde.rs/field-attrs.html#default) like this.

```
#[serde(default)]
kvno: u32,
```

What this does is if the field is *not* present in the configuration file, it automatically is filled with the default value for a u32 (which in this case is zero). 

Alternately, if we need to default to 1, then you can use the `#[serde(default = <name of fn>)]` and that allows you to make a function like:

```
fn default_kvno() -> u32 { 1 }
```

Which will be the default if this isn't set in the config. So it just makes it a bit nicer for the user, they don't need to set the kvno unless it's required, but from our side we always have it available.